Add the 6th tutorial - glm

Yibeichan opened the pull request:
I add the 6th tutorial – conducting glm using pydra. 
This tutorial is converted from the nilearn.glm tutorial
I also create an output folder to save the glm output.

satra commented:
the `.ipynb_checkpoints` folder should be add to gitignore.

Yibeichan commented:
Thanks, I removed all `.ipynb_checkpoints` and added `.gitignore`

djarecka commented:
@Yibeichan - thank you for the new tutorial! :-)

Could you please add some text describing specific steps? Also, it would be nice to add some code to download the results the workflow produces.

Also some specific suggestions:
- please add new libraries to the tutorial requirements
- I believe the content of `outputs/6_glm` was not meant to be committed
- re changes in previous notebooks: I think it always good to "Clear Output" before committing  changes, I think many changes disappears from this PR if you do it.

Yibeichan commented:
Hello @djarecka, thanks for the comment! I made some changes:
1. remove all outputs in notebooks, delete `outputs/6_glm` folder, add code in the notebook to make this directory folder when using Binder
2. add documentation for the tutorial
3. add python packages in `requirement.txt`

Sorry, I haven't figured out how to download results to local from Binder through code... I only found out how to zip them. I can ask you about the code in our meeting tomorrow.
May I ask why we use jupyter notebooks not jupyter lab? It would be easier to (manually) download everything through jupyter lab.

Also, I realize that my branch naming strategy is not good. I'll make it better in my next task.

djarecka commented:
@Yibeichan - I'm not sure if jupyter lab was supported by binder when I started it, but feel free to move the tutorial to jupyter lab if you thing that would be better (but probably in a separate pr).

Can I ask you why do you want download results?

Yibeichan commented:
@djarecka, aha, I thought we need to provide code for users who run this tutorial to download their results. You said `it would be nice to add some code to download the results the workflow produces` in your first comment. Maybe I misunderstand it?

djarecka commented:
@Yibeichan - so sorry, my mistake, I was thinking about to **visualize** the results of the workflow in the notebook...

Yibeichan commented:
Ah I see. No worries. So visualize them means that I manually plot the results, right? Since we need to clear the output,  we can't get anything (results) automatically.
Re jupyter lab, I'll get a new pr for it :)

Yibeichan commented:
hello @djarecka, I added the visualization part to the 6th notebook and some `png` file to folder `figures`  
I'll start a new pr for jupyterlab and hiding the answers :)

djarecka commented:
@Yibeichan - I think the best would be if people can see the figure they produce. There is no need to commit your figures, if the workflows work fine everyone should have the figures (and other files) at the same location.

Yibeichan commented:
hello @djarecka , I see what you mean, but since we don't commit the whole output folder, figures won't show up until people execute the whole workflow.
In what way you want me to present those figures? Give them the Markdown code?

Yibeichan commented:
okay, got it. One last question, the folder structure picture, do you want me to keep it or remove it as well.

djarecka commented:
I think you should add a code cell with `!tree` or `!ls` (please check if `tree` is available in our binder instance), but  you can also keep the figure (or simply copy paste the content of your folder to the markdown cell) for a reference, so people can check that they have everything

djarecka commented:
btw. I don't think there is any reason to create a special function `firstlevel_workflow`, you use it only once, so you can just create the `wf_firstlevel` workflow. You don't  have to change it, but just a comment

djarecka commented on notebooks/6_glm_from_nilearn.ipynb:
```diff
@@ -0,0 +1,705 @@
+    "In this section, we converte major steps into tasks.\n",
+    "Each pydra task can have multiple python functions. We recommand to put those logically more related functions into the same task.\n",
+    "\n",
+    "It is very **important** to keep in mind what adjacent tasks of your current task will be.\n",
+    "1. Your previous task will decide your arguments in the current task\n",
+    "2. Your next task will be impacted by the returns in the current task"
```
```suggestion
    "It is very **important** to keep in mind what output you want to produce in a specific task. Each output can be later used as an input to the following tasks.
```

Yibeichan commented:
okay! I fixed the visualization part in the glm tutorial and removed unnecessary figures, as we discussed.
Re the function `firstlevel_workflow`, I wanted to give users an example of "workflow as a node", which echoes the example in the `4_intro_workflow.ipynb`.

Yibeichan commented:
Ah I see, I guess it's sort of a habit? Organize things into a function makes me feel everything is tidy? But I can change them back, seriously, if people use this tutorial get confused by what I'm doing. I can change it quickly

Yibeichan commented:
Okay, I removed the function and re-ran the tutorial, works well. :)

djarecka commented:
@Yibeichan - could you please merge master. I've fixed the testing, I realized that right now your notebook does not run as a part of github action workflow

Yibeichan commented:
Yes, merged!

Yibeichan commented:
On my side, it says all checks have passed. Where can I see the failing report?

djarecka commented:
@Yibeichan -  you can open the tests outputs, e.g. https://github.com/nipype/pydra-tutorial/runs/7117193727?check_suite_focus=true and test notebooks section or merge the current master

Yibeichan commented:
yayyy, you're correct! adding matplotlib fixed everything. All checks passed

djarecka commented:
looks great, thank you!
